cmd/sign-off-checker/main.go

```diff
@@ -17,129 +17,198 @@ limitations under the License.
 package main
 
 import (
-	"context"
 	"fmt"
 	"log"
+	"net"
 	"net/http"
+	"net/url"
 	"os"
-	"regexp"
+	"time"
 
 	"github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 	"golang.org/x/oauth2"
+
+	"github.com/heptiolabs/sign-off-checker/pkg/register"
+	"github.com/heptiolabs/sign-off-checker/pkg/webhook"
 )
 
-var client *github.Client
-var secret []byte
+// CLI entrypoint
+func main() {
+	rootCmd := &cobra.Command{
+		Use:     "SHARED_SECRET='[...]' GITHUB_TOKEN='[...]' sign-off-checker",
+		Short:   "A GitHub integration to ensure commits have \"Signed-off-by\".",
+		Args:    cobra.NoArgs,
+		PreRunE: func(_ *cobra.Command, _ []string) error { return validate() },
+		RunE:    func(_ *cobra.Command, _ []string) error { return run() },
+	}
 
-var testRE *regexp.Regexp
+	// $SHARED_SECRET
+	viper.BindEnv("sharedSecret", "SHARED_SECRET")
 
-func init() {
-	testRE = regexp.MustCompile(`(?mi)^signed-off-by:`)
-}
+	// $GITHUB_TOKEN
+	viper.BindEnv("githubToken", "GITHUB_TOKEN")
 
-func loggingMiddleware(handler http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		log.Printf("%s %s %s", r.RemoteAddr, r.Method, r.URL)
-		handler.ServeHTTP(w, r)
-	})
-}
+	// --listen / $LISTEN
+	rootCmd.Flags().String(
+		"listen",
+		":8080",
+		"Set HTTP listen `address`",
+	)
+	viper.BindPFlag("listenAddress", rootCmd.Flags().Lookup("listen"))
+	viper.BindEnv("listenAddress", "LISTEN")
+
+	// --autoregister / $AUTOREGISTER_ORGANIZATIONS
+	rootCmd.Flags().StringSlice(
```

Comment:
I've never liked how you have to repeat the flag to get multiple values... For Ark we created our own array/slice flag type that supports comma-separated values. Not sure how much that matters here.

Reply:
It's not documented well but Cobra also supports comma separated style. This also works with `--autoregister Org1,Org2,Org3`

Reply:
Hmm, it didn't the last time we tried it.

Reply:
It looks like only `StringSlice` splits on commas and `StringArray` doesn't (https://github.com/spf13/pflag/pull/90)?